### PR TITLE
Directive handling

### PR DESCRIPTION
**Pull request by porcellus**
Improving directive handling, as discussed in #1855

### CODE REVIEW

File: test/function/samples/module-level-directive/count.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+export var count = 0;
```
Comment (lukastaegert) [inline test/function/samples/module-level-directive/count.js]: I assume this file can be deleted? 😉

**Comment by lukastaegert** [comment]
Also, very nice tests! I am just doing some final checks before merging it.

File: src/ast/nodes/ExpressionStatement.ts [hunk 2]
```diff
@@ -2,6 +2,26 @@ import Statement from './shared/Statement';
 import MagicString from 'magic-string';
 
 export default class ExpressionStatement extends Statement {
+	directive?: string;
+
+	shouldBeIncluded() {
+		if (this.directive && this.directive !== 'use strict') {
+			if (this.parent.type === "Program") {
+				this.module.warn( // This is necessary, because either way (deleting or not) can lead to errors.
```
Comment (lukastaegert) [inline src/ast/nodes/ExpressionStatement.ts]: `shouldBeIncluded` is run for each tree-shaking pass, which means the error is displayed at least twice, possibly more often. The rest of the code looks nice and I like the approach but the proper place to display the warning is to add a member `initialiseNode`, cf. e.g. the warning in `ThisExpression`. `initialiseNode` is run only once and is meant for node specific initialisations that do not depend on the scope being populated.

**Comment by alippai** [comment]
@lukastaegert are you planning releasing this feature or should I expect some delay (I see there is a nice progress on other PRs and issues)

**Comment by lukastaegert** [comment]
I plan on releasing this together with #1816 and possibly others maybe as early as tomorrow. It was just a little late for the previous release 😜

**Comment by alippai** [comment]
Sure, awesome! Keep up the amazing work

**Comment by alippai** [comment]
@lukastaegert thanks!

**Pull request merged**